app/services/bulk/bulk_upload.rb

```diff
@@ -29,15 +35,30 @@ def process
           secondary_address = table.by_row[x]["secondary_address"]
           found1 = false
           found2 = false
-          for user in users do
+          found3 = false
+          found4 = false
+          for user in users 
             if user["email"] == ActiveRecord::Base.connection.quote(email)
               found1 = true
             end
             if user["username"] == ActiveRecord::Base.connection.quote(username)
               found2 = true
             end
           end
-          if (User.find_by(email: email) == nil and User.find_by(username: username) == nil and !found1 and !found2)
+          for currnet_username in currnet_usernames 
+            if username == currnet_username
+              found3 = true
+              break;
+            end
+          end
+
+          for current_email in current_emails 
+            if current_email == email
+              found4 = true
+            end
+          end
+          
+          if !found3 and !found4 and !found1 and !found2
```

Comment:
Kindly use variable names like email_found and username_present for the understanding of code.